crypto/fipsmodule/rsa/rsa_impl.c

```diff
@@ -787,7 +794,7 @@ int rsa_default_private_transform(RSA *rsa, uint8_t *out, const uint8_t *in,
   // than the CRT attack, but there have likely been improvements since 1997.
   //
   // This check is cheap assuming |e| is small; it almost always is.
-  if (rsa->e != NULL) {
+  if (rsa->e != NULL && !BN_is_zero(rsa->e)) {
```

Comment:
Is it necessary to add this additional representation of "e is missing" to the RSA struct? Why not check for e==0 when parsing and then set e to null in the struct? The danger of requiring these additional checks for e==0 here is we would need to make sure we are not missing any, and we need to make sure they are added to any upstream code that is incorporated.

Reply:
Related: do we need to check for e==0 in any of the other places above where we check whether the pointer is null? For example around line 750?

Reply:
I agree that adding an additional notion of "missing" to this logic is a bit of a code smell. In an earlier iteration of this changeset, I tried doing as you suggest by modifying `rsa_asn1.c` to `NULL`ify zero-valued BNs, but there were many places in that portion of the code that assumed RSA struct members to be non-`NULL`. It felt like the change was quickly increasing in scope, so i backtracked.

We should probably examine that approach more comprehensively, though, so I'll work on a branch implementing that approach so we can compare it side-by-side with the current changeset. Will circle back here when that's ready.

Reply:
ok, turns out the changes were pretty straightforward (commit [here](https://github.com/awslabs/aws-lc/compare/main...WillChilds-Klein:null-e-asn1), full diff with `main` [here](https://github.com/awslabs/aws-lc/commit/e50bee977e699327d1414b3b30345af87194a515)). i'm going to clean up some comments/tests and post another revision.

Reply:
ok, new revision posted that doesn't introduce the notion of "0 means absent" to  `rsa.c` or `rsa_impl.c`. formal verification appears to be happy now.